pkg/server/commands/batch_check_command.go

```diff
@@ -129,9 +143,8 @@ func (bq *BatchCheckQuery) Execute(ctx context.Context, params *BatchCheckComman
 			checkQuery := NewCheckCommand(
 				bq.datastore,
 				bq.checkResolver,
-				bq.typesys,
 				WithCheckCommandLogger(bq.logger),
-				WithCacheController(bq.cacheController),
+				WithCheckCommandCache(bq.cacheController, bq.shouldCache, bq.cacheSingleflightGroup, bq.checkCache, bq.maxCheckCacheSize, bq.checkCacheTTL),
```

Comment:
@ewanharris this is the sort of thing I'm looking to avoid by having the reverse ingress.